### ISSUE TITLE
Recommendation system 2nd

### ISSUE THREAD

**Issue by atherdon** [GroceriStar/tensor-something#11]
**G:** I have gone through the Github repos today and could successfully run 'ocr-grocery-parsing'.  As a next step should I work on the improving this model or you have any other task for me.

---

**A:** Please hold on this idea. right now another team member is working on it - and i don't want that you'll do a same thing. 

or maybe i get you wrong? I mean that repo is related to pinterest image OCR algorytm. if you want to work on OCR related to another things - let's create a separated repository for you only

---
**G:** Do you think we need to perform OCR on handwritten characters using deep learning methods? but will it be useful for our project in hand?

---

**A:** To be honest, i'm not sure. Maybe you better tell me.

---

 **G:** I can also work on your idea receipt recognitions. Or do you think it will be good idea to start with recommendation engine. I have some experience with recommendation engines. I previously build a movie recommendation system for my class project. It can be used for grocery shopping as well as recipe recommendation based on flavor or taste he likes. 

And Is there any possibility I can talk to other AI developer so that I can get a grasp on what we are going to do, our future goals and related things.

---
 **A:**  it'll be awesome if we can work on recommendation engine here. But before we start i want to know a detailed plan what we'll do. It can be a very important feature, but it also required a lot of planning work

Yeah, we can discuss our next steps together. You can use a GitHub issues - mention other developers and start a conversation. This why we have open repositories - so everything is available for each team members

**Comment by gowthamkatari** [comment]
I don't see the need in having hand written characters recognition system for our project as we will not have any handwritten data. Everything is digital.

Give me a couple of days time I will come up with a brief plan for recommendation engine if you like it we can think further.

**Comment by gowthamkatari** [comment]
Recommendation systems

Recommend what?
•	Ingredients
•	Groceries
•	Recipes
•	Diet plans

Based on?

•	User purchase history
•	Similar Groceries he purchased/searched in past
•	Groceries he liked/ rated in past
•	Similar user purchases and ratings
•	Recommendation based on a specific diet (ex. Ketogenic diet, vegan diet, vegetarian diet, weight watchers diet etc.)
•	Popular items in that area

What we can do?

1.	We can have a explore tab in the app where he will be recommended items based on above factors. 
2.	If he now searches for cheese he can be recommended cheese products in the order he will most likely be interested in.
3.	If he search for keto diet he will be recommended foods that people can consumed in a ketogenic diet.
4.	On the other hand if we have family budget tab in app. If he enters the compulsory items he wants to purchase, he can be recommended specific products that can be recommended in that budget
5.	He can now add that items to his grocery list or save that item for later.
6.	Also for each item user add to his list. We can have an option to search similar items.




Plan:
As, we don’t have complete data. We can build a prototype model. A simple and robust recommendation engine for a sample data. We build it in a generic fashion such that when we have our database ready by other developers we can easily fit our data to the recommendation engine. We will have outline of the model and try to fit this model for our cleaned data. We have to try to collect data most relevant to our future project.

Implementation:
1.	This is a very brief plan. As this is a very important feature We might have to spend 1 month on planning , research and other Requirements.

Tasks:
Related Data collection 
Tools that can be used
how this can be build
exploring different methods and options.
Collect all resources 
figure out the possibilities and limitations.
Plan for implementation phase.
How this model can be deployed
How this can be fitted into the app
What features we are going to have


2.	Implementation will be done in next 2 months

This is just a brief idea more research is needed.

**Comment by gowthamkatari** [comment]
many startups are trying to build recommendation systems even they don't have data we have to learn more on how they are approaching this problem

**Comment by atherdon** [comment]
perfect. Or brilliant!
@AkshayChoulwar @philippeTUESIGELEC  
Btw @AakashMallik - it was your idea. hope you happy to see a progress here!

**Comment by atherdon** [comment]
@navens23 @puru-kohli @wael44 

Guys, what do you think about this plan?

**Comment by navens23** [comment]
@gowthamkatari  recommendation system is a great idea. it could be a deal breaker in the long term. but when i think from consumer point of view, we need to be building an application that solves an existing problem.

**Comment by navens23** [comment]
@atherdon we can make use of google OCR api as a base and add more hand engineering to make it robust

**Comment by navens23** [comment]
How about shopping grocery items made easy!
One of the existing problem is reading ingredients. There is ton of them! what i suggest is read those texts using OCR, extract texts, Text search with predefined preferences  and match for important labels like Organic/NON-GMO/Transfat.... and letting app to figure out should i spend more time in shopping this or not

**Comment by gowthamkatari** [comment]
@navens23 yes recommendation system is suggested as an very important  add on feature its not a functional requirement for now.

**Comment by puru-kohli** [comment]
All of the above tasks seem valid. Due to so many options available for recommendation system, a lot of research will have to be done. Instead of completely ignoring the task for later development, let's just see what all is available out there.

**Comment by atherdon** [comment]
@gowthamkatari please start with basic preparations for this system. I'll back with detailed answers to you soon

**Comment by gowthamkatari** [comment]
@atherdon ok. I will start. Thanks.

**Comment by gowthamkatari** [comment]
I could build a basic system to recommend items based on the number of times an item is purchased overall. For this I made up a data set with 5 users and 20 products and a total of 100 purchases. 
I will upload the files soon.

The recommendations will the same for every user as it is popularity based model. However, It is important that the recommendations are personalized for each user. I'm working on that.

**Comment by atherdon** [comment]
Hi Gowtham, thank you for letting me know.
It was crazy few weeks, and i remember that i owe you a lot of anwers!
i'm travel to meet my granny, and leave my notes, related to your
questions. will post my answers when come back to home

Thanks,
Arthur
